### PR TITLE
docs:  add doc strings to collect, render & update_env

### PR DESCRIPTION
**Pull request by jdpatt**
Trying to prevent others from some of the digging that I had to do. I copied in doc strings from mkdocstrings and tweaked them slightly.  Helpful to prevent the user from having to dig deeply in mkdocstrings when first getting started.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [0.1.2](https://github.com/mkdocstrings/handler-template/releases/tag/0.1.2) - 2023-01-06
+
+<small>[Compare with 0.1.1](https://github.com/mkdocstrings/handler-template/compare/0.1.1...0.1.2)</small>
+
+### Code Refactoring
+
+- Added doc strings to collect/render to help a user implement their custom feature set.
+
```
Comment (pawamoy) [inline CHANGELOG.md]: Thanks, but you don't have to update the changelog manually, I have an automated action for that :smile: 
Can you simply revert changes in CHANGELOG.md?

File: project/src/mkdocstrings_handlers/{{python_package_import_name}}/handler.py.jinja [hunk 2]
```diff
@@ -46,9 +46,33 @@ class {{ language|title }}Handler(BaseHandler):
...
+
+        In the implementation, you typically call a subprocess that returns JSON, and load that JSON again into
+        a Python dictionary for example, though the implementation is completely free.
+
+        Arguments:
+            identifier: An identifier that was found in a markdown document for which to collect data. For example,
+                in Python, it would be 'mkdocstrings.handlers' to collect documentation about the handlers module.
+                It can be anything that you can feed to the tool of your choice.
+            config: Configuration options for the tool you use to collect data. Typically called "selection" because
+                these options modify how the objects or documentation are "selected" in the source code.
```
Comment (pawamoy) [inline project/src/mkdocstrings_handlers/{{python_package_import_name}}/handler.py.jinja]: mkdocstrings' own docstrings might be outdated. Now we pass every option to both `collect` and render, so there's no distinction of "selection" and "rendering" anymore. Could you update the docstring accordingly?

File: project/src/mkdocstrings_handlers/{{python_package_import_name}}/handler.py.jinja [hunk 3]
```diff
@@ -46,9 +46,33 @@ class {{ language|title }}Handler(BaseHandler):
...
+            Anything you want, as long as you can feed it to the renderer's `render` method.
+        """
         raise CollectionError("Implement me!")
 
     def render(self, data: CollectorItem, config: dict) -> str:  # noqa: D102
+        """Render a template using provided data and configuration options.
+
+        Arguments:
+            data: The data to render that was collected above in `collect()`.
+            config: The rendering options.
```
Comment (pawamoy) [inline project/src/mkdocstrings_handlers/{{python_package_import_name}}/handler.py.jinja]: Same thing here, could you update to say that config contains all the handler's options defined by the user?

**Review by pawamoy — CHANGES_REQUESTED** [review]
That's great, thanks a lot! A few changes requested but overall LGTM :slightly_smiling_face:

File: CHANGELOG.md (hunk 1 above)
Comment (jdpatt) [reply CHANGELOG.md]: Done, yep.  I realized that when I moved on to the mkdocstrings PR.

**Review by pawamoy — APPROVED** [review]
Perfect, thanks a lot!

**Pull request merged**